monitor/index.ts

```diff
@@ -29,21 +37,100 @@ const checkDeath = (() => {
   };
 })();
 
+function alertWhenViewTooHigh(
+  bestBlockNumber: number,
+  targetEmail: string,
+  state: CheckSealFieldState,
+  currentView: U64
+) {
+  if (currentView.gte(state.viewAlertLevel)) {
+    sendNotice(
+      new chainErrors.ViewTooHigh(bestBlockNumber, currentView),
+      targetEmail
+    );
+  }
+}
+
+function alertWhenMultipleNodesSleeping(
+  bestBlockNumber: number,
+  targetEmail: string,
+  state: CheckSealFieldState,
+  sleepingNodeIndices: number[]
+) {
+  const multipleNodesSleeping = sleepingNodeIndices.length > 1;
+  if (multipleNodesSleeping) {
+    state.prevHasProblem = true;
+    sendNotice(
+      new chainErrors.NodeIsSleeping(bestBlockNumber, sleepingNodeIndices),
+      targetEmail
+    );
+  }
+}
+
+function alertWhenAllNodesWakeUp(
+  bestBlockNumber: number,
+  targetEmail: string,
+  state: CheckSealFieldState,
+  sleepingNodeIndices: number[]
+) {
+  const allNodesNowAwake =
+    sleepingNodeIndices.length === 0 && state.prevHasProblem;
+  if (allNodesNowAwake) {
+    state.prevHasProblem = false;
+    sendNotice(new chainErrors.AllNodesAwake(bestBlockNumber), targetEmail);
+  }
+}
+
+function alertWhenNodesSleepingLong(
+  bestBlockNumber: number,
+  targetEmail: string,
+  state: CheckSealFieldState,
+  sleepingNodeIndices: number[]
+) {
+  for (let idx = 0; idx < state.sleepStreak.length; idx++) {
+    if (sleepingNodeIndices.includes(idx)) {
+      state.sleepStreak[idx] += 1;
+    } else {
+      state.sleepStreak[idx] = 0;
+    }
+  }
+  const longTermSleepingIndices = [];
+  for (let idx = 0; idx < state.sleepStreak.length; idx++) {
```

Comment:
How about merging this loop with the above loop?

Reply:
loop merged